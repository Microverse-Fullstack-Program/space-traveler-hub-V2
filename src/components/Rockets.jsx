const Rockets = () => (
  <div className="rockets-list">
    <div className="rocket">Rocket 1</div>
    <div className="rocket">Rocket 1</div>
    <div className="rocket">Rocket 1</div>
  </div>
);

export default Rockets;
