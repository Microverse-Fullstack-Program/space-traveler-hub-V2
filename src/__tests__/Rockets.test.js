import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
// We're using our own custom render function and not RTL's render.
import renderWithProviders from '../utils/test-utils';
import Rockets from '../components/Rockets';
import server from '../mocks/server';

beforeAll(() => server.listen({ onUnhandledRequest: 'bypass' })); // Enable the mocking in tests.
afterEach(() => server.resetHandlers()); // Reset any runtime handlers tests may use.
afterAll(() => server.close()); // Clean up once the tests are done.

describe('Test rockets before fetching API', () => {
  test('Test for Loading... state', () => {
    renderWithProviders(<Rockets />);
    expect(screen.getByText(/Loading.../i)).toBeInTheDocument();
    expect(screen.queryByText(/Reserve Rocket/i)).not.toBeInTheDocument();
    expect(screen.queryByText(/Falcon 1/i)).not.toBeInTheDocument();
  });
});

describe('Test rockets after fetching API', () => {
  test('Test if rockets are rendered after fetching', async () => {
    renderWithProviders(<Rockets />);
    expect(await screen.findAllByText(/Reserve Rocket/i)).toHaveLength(4);
    expect(await screen.findAllByText(/Falcon 1/i)).toHaveLength(2);
    expect(screen.queryByText(/Loading.../i)).not.toBeInTheDocument();
  });
});
