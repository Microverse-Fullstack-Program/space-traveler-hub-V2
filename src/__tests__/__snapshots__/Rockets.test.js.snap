// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket snapshot should render heading correctly 1`] = `
<div
  class="rocket-list"
>
  <p>
    Loading...
  </p>
</div>
`;
